pyanaconda/ui/gui/spokes/subscription.py

```diff
@@ -0,0 +1,769 @@
+
+    def set_registration_controls_sensitive(self, sensitive, include_register_button=True):
+        """Set sensitivity of the registration controls.
+
+        We set these value individually so that the registration status label
+        that is between the controls will not become grayed out due to setting
+        the top level container insensitive.
+        """
+        self._registration_grid.set_sensitive(sensitive)
+        self._options_expander.set_sensitive(sensitive)
+        self._registration_controls_enabled = sensitive
+        self._update_registration_state()
+
+    # authentication related signals
+
+    def on_account_radio_button_toggled(self, radio):
+        if radio.get_active():
+            self.authentication_method = AuthenticationMethod.USERNAME_PASSWORD
+
+    def on_activation_key_radio_button_toggled(self, radio):
+        if radio.get_active():
+            self.authentication_method = AuthenticationMethod.ORG_KEY
+
+    def on_username_entry_changed(self, editable):
+        self.subscription_request.username = editable.get_text()
+
+    def on_password_entry_changed(self, editable):
+        entered_text = editable.get_text()
+        if entered_text:
+            self.enable_password_placeholder(False)
+        self.subscription_request.account_password.set_secret(entered_text)
+
+    def on_organization_entry_changed(self, editable):
+        self.subscription_request.organization = editable.get_text()
+
+    def on_activation_key_entry_changed(self, editable):
+        entered_text = editable.get_text()
+        keys = None
+        if entered_text:
+            self.enable_activation_key_placeholder(False)
+            keys = entered_text.split(',')
+        # keys == None clears keys in the module, so deleting keys
+        # in the keys field will also clear module data on apply()
+        self.subscription_request.activation_keys.set_secret(keys)
+
+    # system purpose related signals
+
+    def on_system_purpose_checkbox_toggled(self, checkbox):
+        self.system_purpose_revealer.set_reveal_child(checkbox.get_active())
+
+    def on_system_purpose_role_combobox_changed(self, combobox):
+        self.system_purpose_data.role = combobox.get_active_id()
+
+    def on_system_purpose_sla_combobox_changed(self, combobox):
+        self.system_purpose_data.sla = combobox.get_active_id()
+
+    def on_system_purpose_usage_combobox_changed(self, combobox):
+        self.system_purpose_data.usage = combobox.get_active_id()
+
+    # HTTP proxy signals
+
+    def on_http_proxy_checkbox_toggled(self, checkbox):
+        self.http_proxy_revealer.set_reveal_child(checkbox.get_active())
+
+    def on_http_proxy_location_entry_changed(self, editable):
+        hostname = ""
+        port = -1  # not set == -1
+        entered_text = editable.get_text()
+        if entered_text:
+            proxy_obj = ProxyString(url=entered_text)
+            hostname = proxy_obj.host
+            port = proxy_obj.port
```

Comment:
I think you want to remove this line. It's `-1` in case of not set and if set then the value is changed in the if below.

Reply:
Yep, that's redundant, lets remove it.